### PR TITLE
[Excel] (Custom functions) Changes all samples (except tutorial) to match new yo office template

### PR DESCRIPTION
**Pull request by mscharlock**
There are a couple exceptions which are taken care of in other PRs 
I also got rid of the much despised =ADD42

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[cb6696b](https://github.com/OfficeDev/office-js-docs-pr/pull/933/commits/cb6696bb8d332b7709087df6b250435b62442e6a)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-best-practices.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-errors.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-errors.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-troubleshooting.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-troubleshooting.md) | :white_check_mark:Succeeded | |
[docs/images/ClockFunction.gif](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/images/ClockFunction.gif) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C29%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904290513313116-933%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=wtJnzqMCf89seqKI32KHHCeLhb2mMNKDnpJkdZdGFPU%3D&st=2019-04-29T05%3A10%3A04Z&se=2019-05-30T05%3A15%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-best-practices.md [hunk 1]
```diff
@@ -26,7 +26,14 @@ As described in the [custom functions overview](custom-functions-overview.md) ar
 The following code sample shows how to do this association. The sample defines the custom function `add` and associates it with the object in the JSON metadata file where the value of the `id` property is **ADD**.
 
 ```js
-function add(first, second){
+/**
+ * Add two numbers
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-best-practices.md]: "Add two numbers" -> "Adds two numbers."

File: docs/excel/custom-functions-best-practices.md [hunk 2]
```diff
@@ -26,7 +26,14 @@ As described in the [custom functions overview](custom-functions-overview.md) ar
...
+ * @customfunction 
+ * @param {number} first First number
+ * @param {number} second Second number
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-best-practices.md]: Consider adding a hyphen between the param names and the description. Also, add periods to the ends of the descriptions.

File: docs/excel/custom-functions-best-practices.md [hunk 3]
```diff
@@ -43,28 +50,35 @@ Keep in mind the following best practices when creating custom functions in your
 
 * Do not change the value of an `id` property in the JSON metadata file after it's been associated with a corresponding JavaScript function name. You can change the function name that end users see in Excel by updating the `name` property within the JSON metadata file, but you should never change the value of an `id` property after it's been established.
 
-* In the JavaScript file, specify all custom function associations in the same location. For example, the following code sample defines two custom functions and then specifies the association information for both functions.
+* In the JavaScript file, specify a custom function association after each function. For example, the following code sample defines two custom functions and each is followed by their accompanying association information.
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-best-practices.md]: Consider adding a second sentence like "This is done by calling the `CustomFunctions.associate` method." Maybe talk about the `associate` parameters too.

File: docs/excel/custom-functions-best-practices.md [hunk 4]
```diff
@@ -43,28 +50,35 @@ Keep in mind the following best practices when creating custom functions in your
...
 
     ```js
-    function add(first, second){
+    /**
+     * Add two numbers
+     * @customfunction 
+     * @param {number} first First number
+     * @param {number} second Second number
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-best-practices.md]: Same commenting comments as above.

File: docs/excel/custom-functions-errors.md [hunk 5]
```diff
@@ -13,6 +14,11 @@ When you build an add-in that defines custom functions, be sure to include error
 In the following code sample, `.catch` will handle any errors that occur previously in the code.
 
 ```js
+/**
+ * Gets a comment from the hypothetical contoso.com/comments API
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-errors.md]: Add a period to the end of this comment.

File: docs/excel/custom-functions-errors.md [hunk 6]
```diff
@@ -13,6 +14,11 @@ When you build an add-in that defines custom functions, be sure to include error
...
+ * @customfunction
+ * @param {number} commentID ID of comment
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-errors.md]: Same hyphen and period comment as above.

File: docs/excel/custom-functions-troubleshooting.md [hunk 7]
```diff
@@ -31,7 +31,14 @@ In your custom functions' script file, you need to associate each custom functio
 The following example shows an add function, followed by the function's name `add` being associated with the corresponding JSON id `ADD`.
 
 ```js
-function add(first, second){
+/**
+ * Add two numbers
+ * @customfunction 
+ * @param {number} first First number
+ * @param {number} second Second number
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-troubleshooting.md]: Same hyphen and period comment as above.

File: docs/excel/custom-functions-overview.md [hunk 8]
```diff
@@ -11,16 +11,29 @@ Custom functions enable developers to add new functions to Excel by defining tho
 
 [!include[Excel custom functions note](../includes/excel-custom-functions-note.md)]
 
-The following illustration shows an end user inserting a custom function into a cell of an Excel worksheet. The `CONTOSO.ADD42` custom function is designed to add 42 to the pair of numbers that the user specifies as input parameters to the function.
+The following illustration shows how you can call a function you've created with JavaScript or Typescript. In this example, the custom function is a clock, designed by the company Contoso. Entering `=CONTOSO.CLOCK` gives you the present time, which updates itself once every second.
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-overview.md]: Consider rewording the first sentence to include a mention of the workbook. "The following illustration shows your workbook calling a function you've created with JavaScript or Typescript."

Comment (AlexJerabek) [inline docs/excel/custom-functions-overview.md]: There's a minor discrepancy between saying "a function you've created" and "a clock, designed by the company Contoso".

File: docs/excel/custom-functions-overview.md [hunk 9]
```diff
@@ -11,16 +11,29 @@ Custom functions enable developers to add new functions to Excel by defining tho
...
+<img alt="animated image showing an end user inserting the CONTOSO.CLOCK custom function into a cell of an Excel worksheet" src="../images/ClockFunction.gif" />
 
-The following code defines the `ADD42` custom function.
+The following code defines the custom function `=CLOCK`.
 
 ```js
-function add42(a, b) {
-  return a + b + 42;
+/**
+ * Displays the current time once a second
```
Comment (AlexJerabek) [inline docs/excel/custom-functions-overview.md]: Add a period to the end of the comment.

File: docs/excel/custom-functions-overview.md (hunk 8 above)
Comment (mscharlock) [reply docs/excel/custom-functions-overview.md]: Good point

File: docs/excel/custom-functions-best-practices.md (hunk 2 above)
Comment (mscharlock) [reply docs/excel/custom-functions-best-practices.md]: Weirdly with the way that this thing works the hyphen then would show in Intellisense so I'm gonna do the periods but not the hyphens

**Comment by o365devx** [comment]
Docs Build status updates of commit _[0dd3f6d](https://github.com/OfficeDev/office-js-docs-pr/pull/933/commits/0dd3f6d8a80b383f458d4cc78965020b76aaabb9)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-best-practices.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-errors.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-errors.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-troubleshooting.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-troubleshooting.md) | :white_check_mark:Succeeded | |
[docs/images/ClockFunction.gif](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/images/ClockFunction.gif) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C29%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904291845526454-933%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=pPGTs4epYvYWxQiQxJb%2BkU3TkiFUNvCutiS9fAMDih0%3D&st=2019-04-29T18%3A42%3A19Z&se=2019-05-30T18%3A47%3A19Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-overview.md [hunk 10]
```diff
@@ -11,24 +11,37 @@ Custom functions enable developers to add new functions to Excel by defining tho
...
-The following code defines the `ADD42` custom function.
+The following code defines the custom function `=CLOCK`.
 
 ```js
-function add42(a, b) {
-  return a + b + 42;
+/**
+ * Displays the current time once a second.
+ * @customfunction
+ * @param {CustomFunctions.StreamingInvocation<string>} invocation Custom function invocation
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]: This parameter type is ugly to see in an overview and makes me wonder if we should have a simpler example for the overview. For now maybe just have a link to the streaming function section for more info about how this function works (so you don't have to get into details about the context. BTW: I think it should always be called "context" not "invocation" or "invocationContext"? something that better describes it came from Excel with more info you can use or set.

File: docs/excel/custom-functions-overview.md [hunk 11]
```diff
@@ -11,24 +11,37 @@ Custom functions enable developers to add new functions to Excel by defining tho
 
 [!include[Excel custom functions note](../includes/excel-custom-functions-note.md)]
 
-The following illustration shows an end user inserting a custom function into a cell of an Excel worksheet. The `CONTOSO.ADD42` custom function is designed to add 42 to the pair of numbers that the user specifies as input parameters to the function.
+The following illustration shows your workbook calling a function you've created with JavaScript or Typescript. In this example, the custom function is a clock. Entering `=CONTOSO.CLOCK` gives you the present time, which updates itself once every second. `CONTOSO` is an optional namespace added to better identify the CLOCK function from possible other CLOCK functions in other add-ins.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]: I thought namespace was required? If not then yes, it should say "optional namespace"

File: docs/excel/custom-functions-overview.md [hunk 12]
```diff
@@ -11,24 +11,37 @@ Custom functions enable developers to add new functions to Excel by defining tho
...
-function add42(a, b) {
-  return a + b + 42;
+/**
+ * Displays the current time once a second.
+ * @customfunction
+ * @param {CustomFunctions.StreamingInvocation<string>} invocation Custom function invocation
+ */
+function clock(invocation) {
+  const timer = setInterval(() => {
+    const time = currentTime();
```
Comment (davidchesnut) [inline docs/excel/custom-functions-overview.md]: This is a helper function. So should also include for completeness the following code from the template
```js
function currentTime() {
  return new Date().toLocaleTimeString();
}
```

**Comment by o365devx** [comment]
Docs Build status updates of commit _[222ea73](https://github.com/OfficeDev/office-js-docs-pr/pull/933/commits/222ea73e085fe1a52d283fd058408a6003b39cb0)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-best-practices.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-errors.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-errors.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-troubleshooting.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-troubleshooting.md) | :white_check_mark:Succeeded | |
[docs/images/ClockFunction.gif](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/images/ClockFunction.gif) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C1%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201905010046364866-933%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=h%2FqWIztJXMHhB%2BFd6nN1Jha8cdhpLRfLV2kPY1gb4s4%3D&st=2019-05-01T00%3A42%3A55Z&se=2019-06-01T00%3A47%3A55Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[7729cd3](https://github.com/OfficeDev/office-js-docs-pr/pull/933/commits/7729cd3d5383a0fd62f6160746d1ea8d19f21815)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-67a482eb)

<a id="67a482eb"></a>
###
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/office-js-docs-pr, working directory: W:\ktrt-s, target branch: master, target branch commit: 296f6364b12a794c886e03de80b6d041786e714f, source commit: 7729cd3d5383a0fd62f6160746d1ea8d19f21815.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C1%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201905010400168313-933%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=IlrfrJ0hk%2B0gMLdXcmykjy5%2FUOadWbzJAcOcrK1BN4c%3D&st=2019-05-01T03%3A55%3A28Z&se=2019-06-01T04%3A00%3A28Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-best-practices.md [hunk 13]
```diff
@@ -35,58 +42,99 @@ CustomFunctions.associate("ADD", add);
...
+            "name": "third",
+            "description": "third optional number to add",
+            "type": "number",
+            "dimensionality": "scalar",
+            "optional": true
         }
-      ]
+    ],
+    "options": {
+        "sync": false
```
Comment (mscharlock) [inline docs/excel/custom-functions-best-practices.md]:
```suggestion
        "sync": false
```
Check but I don't think we have sync anymore

**Comment by o365devx** [comment]
Docs Build status updates of commit _[30b9229](https://github.com/OfficeDev/office-js-docs-pr/pull/933/commits/30b9229062f32aff4cf64d721416b3d37d73f768)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-best-practices.md) | :bulb:Suggestion | | [Details](#user-content-2a32999c-db237510)
[docs/excel/custom-functions-errors.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-errors.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-troubleshooting.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-troubleshooting.md) | :white_check_mark:Succeeded | |
[docs/images/ClockFunction.gif](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/images/ClockFunction.gif) | :white_check_mark:Succeeded | |

<a id="2a32999c-db237510"></a>
### [docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-templatizesamples/docs/excel/custom-functions-best-practices.md)
 - **Line 53**: **[Suggestion]** Illegal link: `&lt;a href=&quot;#custom-functions-json-autogeneration.md&quot;&gt;Create JSON metadata for custom functions&lt;/a&gt;` -- missing bookmark. The file excel/custom-functions-best-practices.md doesn&#39;t contain a bookmark named custom-functions-json-autogeneration.md.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C1%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201905010402204102-933%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=neqZA4idJE%2FXrD8aDEVf5JPTM2ATu3mGWlknON094wQ%3D&st=2019-05-01T03%3A58%3A48Z&se=2019-06-01T04%3A03%3A48Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mscharlock** [comment]
Ok I'm going to merge this with knowledge that d.ts is not up to date and that =CLOCK is still to be fixed.

**Pull request merged**